client/hosting/performance/site-performance.tsx

```diff
@@ -131,6 +133,19 @@ export const SitePerformance = () => {
 		} );
 	};
 
+	const handleRecommendationsFilterChange = ( filter?: string ) => {
+		setRecommendationsFilter( filter );
+		const url = new URL( window.location.href );
+
+		if ( filter ) {
+			url.searchParams.set( 'filter', filter );
+		} else {
+			url.searchParams.delete( 'filter' );
+		}
+
+		window.history.replaceState( {}, '', url.toString() );
```

Comment:
We can do `page.replace` like we're doing in the `PageSelector` prop.

Reply:
I tried page.replace but it somehow affects the scroll into view. When I use it, it doesn't scroll to the recommend section unless I use a timeout which seems odd.

Reply:
Alright 👍🏻

Reply:
Oh, actually, you can try passing the full URL, like we're doing in the `PageSelector` prop.

Reply:
The page.replace totally invalidates the scroll into view. I'm going to merge as is for now